gladier_xpcs/tools/gather_xpcs_metadata.py

```diff
@@ -125,51 +168,91 @@ def gather(dataframe):
     # Generate metadata
     hdf_file = data['hdf_file']
     exp_name = pathlib.Path(hdf_file).name.replace(".hdf", "")
-    metadata = gather(hdf_file)
-    metadata.update({
-            'description': f'{exp_name}: Automated data processing.',
-            'creators': [{'creatorName': '8-ID'}],
-            'publisher': 'Automate',
-            'title': exp_name,
-            'subjects': [{'subject': s} for s in exp_name.split('_')],
-            'publicationYear': f'{datetime.datetime.now().year}',
-            'resourceType': {
-                'resourceType': 'Dataset',
-                'resourceTypeGeneral': 'Dataset'
+    gathered_metadata = gather(hdf_file)
+    dc_metadata = {
+        'descriptions': [{
+            "description": f"{exp_name}: Automated data processing.",
+            "descriptionType": "Other"
+        }],
+        'creators': [{'creatorName': '8-ID'}],
+        'publisher': 'Automate',
+        'titles': [{'title': exp_name}],
+        'subjects': [{'subject': s} for s in exp_name.split('_')],
+        'publicationYear': f'{datetime.datetime.now().year}',
+        'resourceType': {
+            'resourceType': 'Dataset',
+            'resourceTypeGeneral': 'Dataset'
+        },
+        'dates': [
+            {
+                # "date": "2020-09-04T21:08:59.027046Z",
+                # "date": "2024-07-17T16:01:36.595827",
+                "date": datetime.datetime.now().isoformat(),
+                "dateType": "Created"
             }
-        })
+        ],
+        'formats': [],
+        'version': "2",
+    }
     extra_metadata = data.get('metadata', {}) or {}
-    metadata.update(extra_metadata)
+    project_metadata = extra_metadata.copy()
     # Create metadata file
     # Some types have changed between search ingests, and they cause the search ingest
     # to fail. Pop them so we don't get the search error.
     for evil_key in ['exchange.partition_norm_factor']:
-        if evil_key in metadata.keys():
-            metadata.pop(evil_key)
-
-    pilot = data['pilot']
+        if evil_key in project_metadata.keys():
+            project_metadata.pop(evil_key)
 
     # Get root_folder, ex: "/data/2020-1/sanat202002/"
     # All datasets need this info to publish correctly, not having it will raise an exception.
-    root_folder = metadata.get('measurement.instrument.acquisition.root_folder')
-    if root_folder:
-        root_folder = pathlib.Path(root_folder)
+    # /gdata/dm/8IDI/2024-1/zhang202402_2/data/H001_27445_QZ_XPCS_test-01000
+    root_folder = pathlib.Path(gathered_metadata['entry.instrument.bluesky.metadata.dataDir'])
+    # 2024-1/zhang202402_2/data/H001_27445_QZ_XPCS_test-01000
+    relative_folder = root_folder.relative_to('/gdata/dm/8IDI/')
```

Comment:
I'm not sure. You're right this is a brittle system, if the path changes this will break. But it depends on how `entry.instrument.bluesky.metadata.dataDir` is gathered. If the metadata doesn't change between deployments, then it will be fine.